udata/migrations/2025-05-26-migrate-topics-to-elements.py

```diff
@@ -0,0 +1,64 @@
+"""Migrate topic.datasets and topics.reuses to topic.elements"""
+
+import logging
+
+from bson import DBRef, ObjectId
+from mongoengine.connection import get_db
+
+log = logging.getLogger(__name__)
+
+
+def migrate(db):
+    log.info("Processing topics…")
+
+    topics = get_db().topic.find()
+
+    for topic in topics:
+        log.info(f"Processing topic {topic['_id']}…")
+        element_refs = []
+
+        # Convert datasets to TopicElement documents
+        for dataset_id in topic.get("datasets", []):
+            element_id = ObjectId()
+            element_doc = {
+                "_id": element_id,
+                "tags": [],
+                "extras": {},
+                "element": {"_cls": "Dataset", "_ref": DBRef("dataset", dataset_id)},
+            }
+
+            # Insert TopicElement document
+            get_db().topic_element.insert_one(element_doc)
```

Comment:
Could we create TopicElement using the mongoengine document instead?

Reply:
Is it a good idea to mix the paradigms in the same migration?